### PR TITLE
move to pyproject.toml, enable ruff-format and mypy

### PR DESCRIPTION
**Pull request by wolfv**
I removed a few try-imports (for py2 - 3 transition, presumably).

I enabled mypy and made it pass. 

I activated `ruff-format`. 

I moved from `setup.py` to `pyproject.toml` (someone should check that it didn't change anything).

Should we change the authors of this project?

### CODE REVIEW

File: conda_forge_webservices/commands.py [hunk 1]
```diff
@@ -371,23 +391,21 @@ def issue_comment(org_name, repo_name, issue_num, title, comment, comment_id=Non
...
         update_team(org_name, repo_name)
-        message = textwrap.dedent("""
-                Hi! This is the friendly automated conda-forge-webservice.
+        message = textwrap.dedent(
+            f"""
+            Hi! This is the friendly automated conda-forge-webservice.
 
-                I just wanted to let you know that I updated the team with maintainers from %s.
-                """ % default_branch)  # noqa
+            I just wanted to let you know that I updated the team with maintainers from {default_branch}.
```
Comment (wolfv) [inline conda_forge_webservices/commands.py:406]: changed to use a f-string.

File: conda_forge_webservices/tests/test_webapp.py [hunk 2]
```diff
@@ -2,12 +2,9 @@
...
 import os
 import hashlib
-try:
-    from urllib.parse import urlencode
-    import unittest.mock as mock
-except ImportError:
-    from urllib import urlencode
-    import mock
+
+from urllib.parse import urlencode
```
Comment (wolfv) [inline conda_forge_webservices/tests/test_webapp.py:6]: removed try import to make mypy complain less.

**Review by beckermr — CHANGES_REQUESTED** [review]
This pr is way to big. Let's do each thing one at a time in separate prs please.

**Comment by wolfv** [comment]
Superseded by a number of other PRs :)

**Pull request closed without merging**